.github/workflows/run-tests-windows.yml

```diff
@@ -13,14 +13,15 @@ concurrency:
 jobs:
   test-runtime:
     name: Test runtime
-    if: github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch' || ((github.event_name == 'schedule' || github.event_name == 'workflow_call') && github.repository == 'scala-native/scala-native')
-    runs-on: windows-2022
+    runs-on: windows-${{ matrix.win }}
     env:
-      ENABLE_EXPERIMENTAL_COMPILER: true
+      ENABLE_EXPERIMENTAL_COMPILER: ${{ matrix.exp }}
```

Comment:
@LeeTibbert this is the new matrix parameter. i wasn't sure if it had any bearing, so i tried this.